### PR TITLE
Add Tropic VMU board support (Baseboard for Teensy 4.1)

### PR DESCRIPTION
**Pull request by PetervdPerk-NXP**
I'm thrilled to introduce Tropic VMU, my latest open-source hobby project! Tropic is an open-source vehicle management unit hardware design that you can manufacture yourself.
 
Tropic VMU utilizes the Teensy 4.1 consisting of:
 
- NXP MIMXRT1062DVJ6B MCU
- Cortex-M7 600Mhz
- 1024K RAM (512 Tightly coupled)
- 8MB Flash
- 100Mbps Ethernet
- USB HS 480Mbit/s
- SD Card interface
 
Tropic Base board consisting of:
 
- Bosch BMI088 IMU
- Bosch BMM150 Magnetometer
- Bosch BMP388 Barometer
- TDK InvenSense ICM-42688-P IMU
- NXP TJA1051TK CAN Transceiver
- RGB Led
- Buzzer
- Button

You can check out the schematics on GitHub [https://github.com/NXPHoverGames/TropicCommunityVMU/.](https://github.com/NXPHoverGames/TropicCommunityVMU/)

### CODE REVIEW

File: platforms/nuttx/cmake/upload.cmake [hunk 1]
```diff
@@ -77,20 +78,33 @@ endif()
...
 
-add_custom_target(upload
-	COMMAND ${PYTHON_EXECUTABLE} ${PX4_SOURCE_DIR}/Tools/px_uploader.py --port ${serial_ports} ${fw_package}
-	DEPENDS ${fw_package}
-	COMMENT "uploading px4"
-	VERBATIM
-	USES_TERMINAL
-	WORKING_DIRECTORY ${PX4_BINARY_DIR}
-	)
+if(PX4_BOARD STREQUAL "nxp_tropic")
```
Comment (dagar) [inline platforms/nuttx/cmake/upload.cmake:81]: hmm, let's find a better place for this board specific bit.

How about carrying these targets in `boards/nxp/tropic/cmake/upload.cmake`?

If teensy variants become a thing we could find somewhere shared.

File: boards/nxp/tropic/firmware.prototype [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+    "board_id": 34,
```
Comment (davids5) [inline boards/nxp/tropic/firmware.prototype:2]: Reserved "NXP ucans32k146"             34
https://github.com/PX4/PX4-Bootloader/blob/main/board_types.txt#L63

**Review by davids5 — CHANGES_REQUESTED** [review]
Please have a pass on the the Version code and board_id. I do not see a Prefix or version setting and yet there is logic `if ver hwtypecmp V6XRT009000 V6XRT010000`  - but this is not an V6XRT

File: boards/nxp/tropic/firmware.prototype (hunk 2 above)
Comment (PetervdPerk-NXP) [reply boards/nxp/tropic/firmware.prototype:2]: So this doesn't implement a px4 bootloader, the teensy bootloader is used.

I suppose I can just remove `firmware.prototype`

`if ver hwtypecmp V6XRT009000 V6XRT010000`

Needs to be removed indeed.

File: boards/nxp/tropic/src/autoleds.c [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+/*
+ * This module shall be used during board bring up of Nuttx.
+ *
+ * The NXP FMUK66-V3 has a separate Red, Green and Blue LEDs driven by the K66
```
Comment (davids5) [inline boards/nxp/tropic/src/autoleds.c:39]: Is this the real LED usage?

**Review by davids5 — CHANGES_REQUESTED** [review]
@PetervdPerk-NXP 

What does `ver all` show for this board? 
Are the Led's usage in auto-Led correct?

File: boards/nxp/tropic/src/autoleds.c (hunk 3 above)
Comment (PetervdPerk-NXP) [reply boards/nxp/tropic/src/autoleds.c:39]: On startup it's used but once `rgbled_pwm` start the mux changes to the PWM timers.

https://github.com/PX4/PX4-Autopilot/blob/76336eac4e94980f7417394f865ffd8d2905fcd1/boards/nxp/tropic/init/rc.board_defaults#L28

**Comment by PetervdPerk-NXP** [comment]
> @PetervdPerk-NXP
> 
> What does `ver all` show for this board? Are the Led's usage in auto-Led correct?
```
nsh> ver all
HW arch: NXP_TROPIC
PX4 git-hash: 76336eac4e94980f7417394f865ffd8d2905fcd1
PX4 version: 1.15.0 80 (17760384)
PX4 git-branch: tropic
OS: NuttX
OS version: Release 12.5.1 (201654783)
OS git-hash: 6fbb26eb521999844f099ac93974fdc7ccca6016
Build datetime: Jun 12 2024 21:34:03
Build uri: localhost
Build variant: default
Toolchain: GNU GCC, 9.3.1 20200408 (release)
PX4GUID: 000500000000000000006147a8053e0b29d7
MCU: i.MX RT1062 r1.1, rev. 1
nsh>
```

File: platforms/nuttx/CMakeLists.txt [hunk 4]
```diff
@@ -297,6 +297,29 @@ else()
...
 
+# create .hex if configured in NuttX
+if (CONFIG_INTELHEX_BINARY)
+
+	string(REPLACE ".elf" ".hex" hex_package ${PX4_BINARY_DIR}/${FW_NAME})
+
+	add_custom_command(
+		OUTPUT ${hex_package}
+		COMMAND
+			arm-none-eabi-objcopy -O ihex
```
Comment (MaEtUgR) [inline platforms/nuttx/CMakeLists.txt:308]: Dev call @dagar mentioned this command should be taken from the CMake detection.

Comment (PetervdPerk-NXP) [reply platforms/nuttx/CMakeLists.txt]: Thanks using `${CMAKE_OBJCOPY}` now instead.

**Review by dagar — APPROVED** [review]

**Pull request merged**